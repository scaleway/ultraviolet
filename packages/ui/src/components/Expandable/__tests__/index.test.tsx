import { renderWithTheme } from '@utils/test'
import { describe, expect, test } from 'vitest'
import { Expandable } from '..'

describe('expandable', () => {
  test('renders correctly with default values', () => {
    const { asFragment } = renderWithTheme(
      <Expandable>Sample Expandable</Expandable>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders correctly opened', () => {
    const { asFragment } = renderWithTheme(
      <Expandable opened>Sample Expandable</Expandable>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders correctly with minHeight', () => {
    const { asFragment } = renderWithTheme(
      <Expandable minHeight={5}>Sample Expandable</Expandable>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders correctly with className', () => {
    const { asFragment } = renderWithTheme(
      <Expandable className="test">Sample Expandable</Expandable>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders correctly with animationDuration', () => {
    const { asFragment } = renderWithTheme(
      <Expandable animationDuration={500}>Sample Expandable</Expandable>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders correctly with animationDuration set to 0', () => {
    const { asFragment } = renderWithTheme(
      <Expandable animationDuration={0}>Sample Expandable</Expandable>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
