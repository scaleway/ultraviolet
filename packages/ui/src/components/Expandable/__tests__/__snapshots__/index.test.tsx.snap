// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`expandable > renders correctly opened 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__mlltm91"
      data-is-animated="false"
      style="--mlltm90: 300ms;"
    >
      Sample Expandable
    </div>
  </div>
</DocumentFragment>
`;

exports[`expandable > renders correctly with animationDuration 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__mlltm91"
      data-is-animated="false"
      style="--mlltm90: 500ms; max-height: 0px; overflow: hidden;"
    >
      Sample Expandable
    </div>
  </div>
</DocumentFragment>
`;

exports[`expandable > renders correctly with animationDuration set to 0 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      style="display: none;"
    >
      Sample Expandable
    </div>
  </div>
</DocumentFragment>
`;

exports[`expandable > renders correctly with className 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="test styles__mlltm91"
      data-is-animated="false"
      style="--mlltm90: 300ms; max-height: 0px; overflow: hidden;"
    >
      Sample Expandable
    </div>
  </div>
</DocumentFragment>
`;

exports[`expandable > renders correctly with default values 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__mlltm91"
      data-is-animated="false"
      style="--mlltm90: 300ms; max-height: 0px; overflow: hidden;"
    >
      Sample Expandable
    </div>
  </div>
</DocumentFragment>
`;

exports[`expandable > renders correctly with minHeight 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__mlltm91"
      data-is-animated="false"
      style="--mlltm90: 300ms; max-height: 5px; overflow: hidden;"
    >
      Sample Expandable
    </div>
  </div>
</DocumentFragment>
`;
